### PR TITLE
User will not lose components when change they position in the new D&D

### PR DESCRIPTION
**Pull request by YaroslavOvdii**
@sofiiakvasnevska 

## Issue
https://github.com/Fliplet/fliplet-studio/issues/5354

## Description
Changed components so they work with Fliplet.Widget.instance now.

## Screenshots/screencasts
https://share.getcloudapp.com/Qwu7RnDj

## Backward compatibility

This change is fully backward compatible.

### CODE REVIEW

File: css/build.css [hunk 1]
```diff
@@ -163,3 +163,8 @@ div[data-state="touchID"] button {
 	vertical-align: top;
 	color: #e9376b;
 }
+
+/* Fix component view when it is placed in the container */
+[data-widget-package="com.fliplet.lock"] {
+	width: 100%;
```
Comment (tonytlwu) [inline css/build.css]:
```suggestion
  width: 100%;
```

File: build.html [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<div class="passcode-wrapper" data-id="{{id}}" data-uuid="{{uuid}}">
+<div class="passcode-wrapper" data-passcode-wrapper-id="{{id}}" data-uuid="{{uuid}}">
```
Comment (tonytlwu) [inline build.html]:
```suggestion
<div class="passcode-wrapper" data-passcode-id="{{id}}" data-passcode-uuid="{{uuid}}">
```

**Review by tonytlwu — CHANGES_REQUESTED** [review]
See comments

File: css/build.css [hunk 3]
```diff
@@ -163,3 +163,8 @@ div[data-state="touchID"] button {
 	vertical-align: top;
 	color: #e9376b;
 }
+
+/* Fix component view when it is placed in the container */
+[data-widget-package="com.fliplet.lock"] {
+  width: 100%;
```
Comment (tonytlwu) [inline css/build.css:169]: I'm not going to approve this until @hcarneiro answers the question at https://github.com/Fliplet/fliplet-widget-fliplet-login/pull/50/files#r386035704

File: js/build.js [hunk 4]
```diff
@@ -1,12 +1,10 @@
-$('.passcode-wrapper').each(function() {
+Fliplet.Widget.instance('passcode', function(data){
   var $lock = $(this);
-  var widgetId = $lock.data('id');
-  var widgetUuid = $lock.data('uuid');
+  var widgetId = data.id;
+  var widgetUuid = data.uuid;
   if (!widgetId) {
-    return;
+      return;
```
Comment (tonytlwu) [inline js/build.js]:
```suggestion
    return;
```

**Review by tonytlwu — CHANGES_REQUESTED** [review]

File: build.html [hunk 5]
```diff
@@ -1,4 +1,4 @@
-<div class="passcode-wrapper" data-id="{{id}}" data-uuid="{{uuid}}">
+<div class="passcode-wrapper" data-passcode-id="{{id}}" data-passcode-uuid="{{uuid}}">
```
Comment (tonytlwu) [inline build.html]: I realised that calling it `data-lock-id` would make better sense, inline with the `com.fliplet.lock` name.

```suggestion
<div class="passcode-wrapper" data-lock-id="{{id}}" data-lock-uuid="{{uuid}}">
```

Everything else is good. I'll approve after this change.

**Review by tonytlwu — CHANGES_REQUESTED** [review]

File: build.html (hunk 5 above)
Comment (YaroslavOvdii) [reply build.html]: @tonytlwu Added this changes

**Review by tonytlwu — APPROVED** [review]
@sofiiakvasnevska This is ready for testing

Risk: Low
Affected: Lock components in apps
Verify: Component works as usual

**Pull request merged**